src/auth/strategy/apikey.strategy.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { PassportStrategy } from '@nestjs/passport';
+import { HeaderAPIKeyStrategy } from 'passport-headerapikey';
+
+import { AuthService } from '../auth.service';
+
+@Injectable()
+export class ApiKeyStrategy extends PassportStrategy(HeaderAPIKeyStrategy, 'x-api-key') {
+  constructor(private authService: AuthService, private configService: ConfigService) {
```

Comment:
we can make the `authService` and `configService` readonly

Reply:
this is updated now